L1Trigger/TrackFindingTracklet/src/TrackletLUT.cc

```diff
@@ -0,0 +1,945 @@
+    double rratio2 = rmean / (r - 0.5 * dr);
+
+    double z1 = (z - 0.5 * dz) * rratio1 + z0cut * (rratio1 - 1.0);
+    double z2 = (z + 0.5 * dz) * rratio1 + z0cut * (rratio1 - 1.0);
+    double z3 = (z - 0.5 * dz) * rratio2 + z0cut * (rratio2 - 1.0);
+    double z4 = (z + 0.5 * dz) * rratio2 + z0cut * (rratio2 - 1.0);
+    double z5 = (z - 0.5 * dz) * rratio1 - z0cut * (rratio1 - 1.0);
+    double z6 = (z + 0.5 * dz) * rratio1 - z0cut * (rratio1 - 1.0);
+    double z7 = (z - 0.5 * dz) * rratio2 - z0cut * (rratio2 - 1.0);
+    double z8 = (z + 0.5 * dz) * rratio2 - z0cut * (rratio2 - 1.0);
+
+    double zmin = std::min({z1, z2, z3, z4, z5, z6, z7, z8});
+    double zmax = std::max({z1, z2, z3, z4, z5, z6, z7, z8});
+
+    int NBINS = settings_.NLONGVMBINS() * settings_.NLONGVMBINS();
+
+    int zbin1 = NBINS * (zmin + settings_.zlength()) / (2 * settings_.zlength());
+    int zbin2 = NBINS * (zmax + settings_.zlength()) / (2 * settings_.zlength());
+
+    if (zbin1 >= NBINS)
+      return -1;
+    if (zbin2 < 0)
+      return -1;
+
+    if (zbin2 >= NBINS)
+      zbin2 = NBINS - 1;
+    if (zbin1 < 0)
+      zbin1 = 0;
+
+    // This is a 10 bit word:
+    // xxx|yyy|z|rrr
+    // xxx is the delta z window
+    // yyy is the z bin
+    // z is flag to look in next bin
+    // rrr first fine z bin
+    // NOTE : this encoding is not efficient z is one if xxx+rrr is greater than 8
+    //        and xxx is only 1,2, or 3
+    //        should also reject xxx=0 as this means projection is outside range
+
+    int value = zbin1 / 8;
+    value *= 2;
+    if (zbin2 / 8 - zbin1 / 8 > 0)
+      value += 1;
+    value *= 8;
+    value += (zbin1 & 7);
+    assert(value / 8 < 15);
+    int deltaz = zbin2 - zbin1;
+    if (deltaz > 7) {
+      deltaz = 7;
+    }
+    assert(deltaz < 8);
+    value += (deltaz << 7);
+
+    return value;
+
+  } else {
+    if (std::abs(z) < 2.0 * z0cut)
+      return -1;
+
+    double zmean = settings_.zmean(layerdisk - N_LAYER);
+    if (z < 0.0)
+      zmean = -zmean;
+
+    double r1 = (r + 0.5 * dr) * (zmean + z0cut) / (z + 0.5 * dz + z0cut);
+    double r2 = (r - 0.5 * dr) * (zmean - z0cut) / (z + 0.5 * dz - z0cut);
+    double r3 = (r + 0.5 * dr) * (zmean + z0cut) / (z - 0.5 * dz + z0cut);
+    double r4 = (r - 0.5 * dr) * (zmean - z0cut) / (z - 0.5 * dz - z0cut);
+    double r5 = (r + 0.5 * dr) * (zmean - z0cut) / (z + 0.5 * dz - z0cut);
+    double r6 = (r - 0.5 * dr) * (zmean + z0cut) / (z + 0.5 * dz + z0cut);
+    double r7 = (r + 0.5 * dr) * (zmean - z0cut) / (z - 0.5 * dz - z0cut);
+    double r8 = (r - 0.5 * dr) * (zmean + z0cut) / (z - 0.5 * dz + z0cut);
+
+    double rmin = std::min({r1, r2, r3, r4, r5, r6, r7, r8});
+    double rmax = std::max({r1, r2, r3, r4, r5, r6, r7, r8});
+
+    int NBINS = settings_.NLONGVMBINS() * settings_.NLONGVMBINS() / 2;
+
+    double rmindisk = settings_.rmindiskvm();
+    double rmaxdisk = settings_.rmaxdiskvm();
+
+    if (iseed == 6)
```

Comment:
@aryd could these use the new seed enum too?

Reply:
Yes - have changed, these were missed.